pkg/controller/mpi_job_controller.go

```diff
@@ -656,13 +657,14 @@ func (c *MPIJobController) syncHandler(key string) error {
 				return err
 			}
 		}
-		if mpiJob.Spec.MPIImplementation == kubeflow.MPIImplementationIntel ||
+		// If we want to run process in launcher, we should create a service for launcher.
```

Comment:
The existing Service created with selector `training.kubeflow.org/job-role: worker` which can't be used directly. Without creating Service for launcher, we should change the service for worker and remove the selector.
Create a service is more preferable for me. WDYT @tenzen-y @alculquicondor

Reply:
I think it's more efficient to just change the selector.

Reply:
I would just change it for all

Reply:
@alculquicondor So it's a refactor of the service mechanism so that we create a job level service for all, no service for launcher and service for workers any more. Then we will use the job level service to access the target pod when necessary. Is that right ?

Reply:
Yes. Although I'm not sure if IntelMPI would work with just one Service, as it's very picky about the hostname. Worth trying.

Reply:
OK,I will try it. While we just make the service refactor in this PR or with another one, since it's already a way to run with respect to the original service design?

Reply:
here should be fine

Reply:
SGTM

Reply:
I've add a service for both launcher and workers, for IntelMPI which need to access launcher with hostname, I modify the searches part in DNSConfig.